core/src/main/java/it/pagopa/selfcare/mscore/core/OnboardingServiceImpl.java

```diff
@@ -101,4 +106,24 @@ public Institution persistOnboarding(String institutionId, String
         }
     }
 
+    @Override
+    public void deleteOnboardedInstitution(String institutionId, String productId) {
+
+        log.trace("persist logic delete Onboarding - start");
+        Onboarding onboarding = new Onboarding();
+        onboarding.setStatus(RelationshipState.DELETED);
+        onboarding.setProductId(productId);
+        institutionConnector.findAndRemoveOnboarding(institutionId, onboarding);
```

Comment:
it might be better to DELETE only an onboarding in status ACTIVE....if an onboarding in other states exists (for. SUSPEND or another DELETED), the method will be updated all onboarding for that product